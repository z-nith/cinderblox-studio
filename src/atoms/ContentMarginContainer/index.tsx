import ContentMarginContainer from './ContentMarginContainer';

export default ContentMarginContainer;
