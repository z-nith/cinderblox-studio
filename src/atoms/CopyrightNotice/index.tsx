import CopyrightNotice from './CopyrightNotice';

export default CopyrightNotice;
