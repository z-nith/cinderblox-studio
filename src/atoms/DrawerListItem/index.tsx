import React from 'react';
import DrawerListItem, { DrawerListItemPropTypes } from './DrawerListItem';
import { ThemeProvider } from '@material-ui/core';
import { alternateDarkTheme } from 'theme';

const Index = (props: DrawerListItemPropTypes) => {
    return (
        <ThemeProvider theme={alternateDarkTheme}>
            <DrawerListItem {...props} />
        </ThemeProvider>
    );
};

export default Index;
