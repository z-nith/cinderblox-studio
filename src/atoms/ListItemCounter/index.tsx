import ListItemCounter from './ListItemCounter';

export default ListItemCounter;
