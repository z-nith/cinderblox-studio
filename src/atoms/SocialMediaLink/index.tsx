import SocialMediaLink from './SocialMediaLink';

export default SocialMediaLink;
