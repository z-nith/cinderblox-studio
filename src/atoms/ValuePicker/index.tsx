import ValuePicker from './ValuePicker';

export default ValuePicker;
