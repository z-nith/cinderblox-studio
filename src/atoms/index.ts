import AppBar from './AppBar';
import Avatar from './Avatar';
import BaseLink from './BaseLink';
import Button from './Button';
import CopyrightNotice from './CopyrightNotice';
import { GridContainer, GridItem } from './Grid';
import Link from './Link';
import ListItemCounter from './ListItemCounter';
import ScrollToTopButton from './ScrollToTopButton';
import SocialMediaLink from './SocialMediaLink';
import TextField from './TextField';
import Tooltip from './Tooltip';
import Typography from './Typography';
import ValuePicker from './ValuePicker';

export {
    ListItemCounter,
    ScrollToTopButton,
    Button,
    BaseLink,
    CopyrightNotice,
    GridContainer,
    GridItem,
    Link,
    SocialMediaLink,
    TextField,
    ValuePicker,
    Typography,
    Avatar,
    Tooltip,
    AppBar,
};
