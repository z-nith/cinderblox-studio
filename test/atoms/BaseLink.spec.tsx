import React from 'react';
import { BaseLink } from 'atoms';
import { mount } from 'enzyme';
import { mockNextUseRouter } from '../mocks';

describe(`Link`, () => {
    // Mocks Next.js route
    mockNextUseRouter({
        route: '/',
        pathname: '/',
        query: '',
        asPath: `/`,
    });

    it(`should render without error`, () => {
        const wrapper = mount(<BaseLink href={'/'}>Hello</BaseLink>);
        expect(wrapper).toMatchSnapshot();
    });
});
