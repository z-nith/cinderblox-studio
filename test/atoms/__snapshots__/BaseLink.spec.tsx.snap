// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render without error 1`] = `
<ForwardRef
  href="/"
>
  <BaseLink
    href="/"
    innerRef={null}
  >
    <WithStyles(ForwardRef(Link))
      className="active"
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {},
          "render": [Function],
        }
      }
      href="/"
    >
      <ForwardRef(Link)
        className="active"
        classes={
          Object {
            "button": "MuiLink-button",
            "focusVisible": "Mui-focusVisible",
            "root": "MuiLink-root",
            "underlineAlways": "MuiLink-underlineAlways",
            "underlineHover": "MuiLink-underlineHover",
            "underlineNone": "MuiLink-underlineNone",
          }
        }
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {},
            "render": [Function],
          }
        }
        href="/"
      >
        <WithStyles(ForwardRef(Typography))
          className="MuiLink-root MuiLink-underlineHover active"
          color="primary"
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {},
              "render": [Function],
            }
          }
          href="/"
          onBlur={[Function]}
          onFocus={[Function]}
          variant="inherit"
        >
          <ForwardRef(Typography)
            className="MuiLink-root MuiLink-underlineHover active"
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter",
                "alignJustify": "MuiTypography-alignJustify",
                "alignLeft": "MuiTypography-alignLeft",
                "alignRight": "MuiTypography-alignRight",
                "body1": "MuiTypography-body1",
                "body2": "MuiTypography-body2",
                "button": "MuiTypography-button",
                "caption": "MuiTypography-caption",
                "colorError": "MuiTypography-colorError",
                "colorInherit": "MuiTypography-colorInherit",
                "colorPrimary": "MuiTypography-colorPrimary",
                "colorSecondary": "MuiTypography-colorSecondary",
                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                "displayBlock": "MuiTypography-displayBlock",
                "displayInline": "MuiTypography-displayInline",
                "gutterBottom": "MuiTypography-gutterBottom",
                "h1": "MuiTypography-h1",
                "h2": "MuiTypography-h2",
                "h3": "MuiTypography-h3",
                "h4": "MuiTypography-h4",
                "h5": "MuiTypography-h5",
                "h6": "MuiTypography-h6",
                "noWrap": "MuiTypography-noWrap",
                "overline": "MuiTypography-overline",
                "paragraph": "MuiTypography-paragraph",
                "root": "MuiTypography-root",
                "srOnly": "MuiTypography-srOnly",
                "subtitle1": "MuiTypography-subtitle1",
                "subtitle2": "MuiTypography-subtitle2",
              }
            }
            color="primary"
            component={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "propTypes": Object {},
                "render": [Function],
              }
            }
            href="/"
            onBlur={[Function]}
            onFocus={[Function]}
            variant="inherit"
          >
            <ForwardRef(NextComposed)
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover active MuiTypography-colorPrimary"
              href="/"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <Link
                href="/"
              >
                <a
                  className="MuiTypography-root MuiLink-root MuiLink-underlineHover active MuiTypography-colorPrimary"
                  href="/"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                >
                  Hello
                </a>
              </Link>
            </ForwardRef(NextComposed)>
          </ForwardRef(Typography)>
        </WithStyles(ForwardRef(Typography))>
      </ForwardRef(Link)>
    </WithStyles(ForwardRef(Link))>
  </BaseLink>
</ForwardRef>
`;
